libraries/workmanager/impl/src/main/kotlin/io/element/android/libraries/workmanager/impl/DefaultWorkManagerScheduler.kt

```diff
@@ -51,6 +52,18 @@ class DefaultWorkManagerScheduler(
         )
     }
 
+    override fun hasPendingWork(sessionId: SessionId, requestType: WorkManagerRequestType): Boolean {
+        val workInfos = workManager.getWorkInfosByTag(workManagerTag(sessionId, requestType)).get().orEmpty()
+        return workInfos.any { info ->
+            val isPeriodic = info.periodicityInfo != null
+            val isCancelled = info.state == WorkInfo.State.CANCELLED
+            // It has pending work if:
+            // - It's not periodic and is not finished.
+            // - It's periodic and is not cancelled - since it'll be run again in a next iteration otherwise
+            !isPeriodic && !info.state.isFinished || isPeriodic && !isCancelled
```

Comment:
In case of failure of the worker, for instance when it returns `Result.failure()`, will `!isCancelled` return `true`?